### DISCUSSION TITLE
Multiple IKE SAs between same peer

### DISCUSSION THREAD

**Discussion by Jaditya-89** [strongswan/strongswan#2083] [General]
Hi Strongswan experts, 

I have two peers both initiating IKE SA connection with start_action = start on both peers. I expect two IKE SAs to be created on each peer. But sometimes I am seeing more than 2 entries. The config is configured programatically and SA establishment is triggered around same time. This is happening at around scale of 1k sessions and happening with multiple pair of peers.
For example, 

```
73373600-0000-0000-0000-000000000000: #9716, ESTABLISHED, IKEv2, 36e727df84b5f7da_i 89d87420dd65fdbe_r*
  local  'abcd' @ 128.0.0.148[4500]
  remote 'efgh' @ 128.0.0.149[4500]
  AES_CBC-256/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_4096
  established 962s ago, rekeying in 24981s, reauth in 54916s
  73373600-0000-0000-0000-000000000000: #8355, reqid 796, INSTALLED, TRANSPORT-in-UDP, ESP:AES_GCM_16-256
    installed 962s ago, rekeying in 8781s, expires in 10918s
    in  c435f300,      0 bytes,     0 packets
    out c2fad35a,      0 bytes,     0 packets
    local  128.0.0.148/32
    remote 128.0.0.149/32
73373600-0000-0000-0000-000000000000: #7852, ESTABLISHED, IKEv2, fe092a37b66a3f82_i* e4ed49d271d62b22_r
  local  'abcd' @ 128.0.0.148[4500]
  remote 'efgh' @ 128.0.0.149[4500]
  AES_CBC-256/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_4096
  established 926s ago, rekeying in 26788s, reauth in 55694s
  73373600-0000-0000-0000-000000000000: #8405, reqid 81, INSTALLED, TRANSPORT-in-UDP, ESP:AES_GCM_16-256
    installed 926s ago, rekeying in 9196s, expires in 10954s
    in  c16ca536,      0 bytes,     0 packets
    out c45d94ec,      0 bytes,     0 packets
    local  128.0.0.148/32
    remote 128.0.0.149/32
73373600-0000-0000-0000-000000000000: #7206, ESTABLISHED, IKEv2, b28c2a0dcf4ac7f7_i* 8fd8c6f29a648eed_r
  local  'abcd' @ 128.0.0.148[4500]
  remote 'efgh' @ 128.0.0.149[4500]
  AES_CBC-256/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_4096
  established 933s ago, rekeying in 27024s, reauth in 55736s
  73373600-0000-0000-0000-000000000000: #8384, reqid 81, INSTALLED, TRANSPORT-in-UDP, ESP:AES_GCM_16-256
    installed 934s ago, rekeying in 9522s, expires in 10947s
    in  c154a331,      0 bytes,     0 packets
    out c4b96107,      0 bytes,     0 packets
    local  128.0.0.148/32
    remote 128.0.0.149/32
73373600-0000-0000-0000-000000000000: #7027, ESTABLISHED, IKEv2, ad8cb443291751da_i* a98d12bb8114a338_r
  local  'abcd' @ 128.0.0.148[4500]
  remote 'efgh' @ 128.0.0.149[4500]
  AES_CBC-256/HMAC_SHA2_512_256/PRF_HMAC_SHA2_512/MODP_4096
  established 936s ago, rekeying in 27827s, reauth in 54950s
  73373600-0000-0000-0000-000000000000: #8379, reqid 81, INSTALLED, TRANSPORT-in-UDP, ESP:AES_GCM_16-256
    installed 936s ago, rekeying in 8939s, expires in 10944s
    in  cf5e6789,      0 bytes,     0 packets
    out c3b35cb0,      0 bytes,     0 packets
    local  128.0.0.148/32
    remote 128.0.0.149/32
```

  I am not seeing this issue everytime . Also, the established times are almost similar but rekeying times are differing(and not adding upto rekey_time = 8 hrs set on both peers).  I am naive in IPsec understanding, so any idea on why this could be happening or how to mitigate this would be very helpful for me.
  Thanks

**Comment by tobiasbrunner** [comment]
Read the logs is all I can say.

**Reply by Jaditya-89** [reply]
Hi @tobiasbrunner , I was able to understand what was happening. Actually , in our solution we don't send  delete message to be received on other peer (using force option here). This delete is sent separately on both peers. But in this case we send delete on one peer only. So other peer has still two entry for IKE_SAs and i have set dpdaction = restart. So, the dpd action will trigger which will try to initiate the connections to the first peer and established IKE SAs will go to CONNECTING state. Now, on the first peer, I load the connection file again (startaction = start ). So, this peer will also initiate connection to second peer. Also, there are two IKE_SA_INIT messages from second peer. So, we will see 3 IKE SA getting established.

Please correct if I am wrong in my understanding. Also would like to know if there is any way I could avoid this issue as this case can lead to multiple IKE SA creation
